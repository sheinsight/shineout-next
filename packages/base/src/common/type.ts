import React, { CSSProperties } from 'react';

export interface CommonType {
  status?: 'error';
  className?: string;
  style?: CSSProperties;
  /**
   * @en There are three built-in size: small、default、large.
   * @cn 不同尺寸
   * @override union
   * @default 'default'
   */
  size?: 'small' | 'large' | 'default';
  /**
   * @en inner title
   * @cn 内嵌标题
   */
  innerTitle?: React.ReactNode;
  /**
   * @en Placeholder title, which needs to be used together with innerTitle
   * @cn 占位标题，需要配合 innerTitle 一起使用
   */
  placeTitle?: React.ReactNode;
}
