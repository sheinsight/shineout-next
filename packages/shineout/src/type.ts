export * as Form from './form/interface';
export * as Input from './input/interface';
export * as Radio from './radio/interface';
export * as Checkbox from './checkbox/interface';
export * as Textarea from './textarea/interface';
export * as Dropdown from './dropdown/interface';
export * as Image from './image/interface';
export * as Spin from './spin/interface';
export * as Icon from './icon/interface';
export * as Switch from './switch/interface';
export * as Popover from './popover/interface';
export * as Tooltip from './tooltip/interface';
export * as EditableArea from './editable-area/interface';
export * as DatePicker from './date-picker/interface';
export * as Divider from './divider/interface';
export * as Message from './message/interface';
