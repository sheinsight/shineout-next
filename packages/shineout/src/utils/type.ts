export type RequireAWithB<T, A extends keyof T, B extends keyof T> = 
  | (Omit<T, A> & { [K in A]?: never })
  | (T & Required<Pick<T, B>>)
  | (T & Required<Pick<T, A | B>>)