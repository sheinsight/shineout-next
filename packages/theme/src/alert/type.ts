// 此文件由脚本自动生成，请勿直接修改。
// This file was generated automatically by a script. Please do not modify it directly.

export interface AlertTokens {
  /**
   * @type {string}
   * @token Brand-1
   * @description 提示框提示色背景颜色
   */
  alertInfoBackgroundColor: string;
  /**
   * @type {string}
   * @token Brand-2
   * @description 提示框提示色边框颜色
   */
  alertInfoBorderColor: string;
  /**
   * @type {string}
   * @token Brand-6
   * @description 提示框提示色字体颜色
   */
  alertInfoFontColor: string;
  /**
   * @type {string}
   * @token Success-1
   * @description 提示框成功色背景颜色
   */
  alertSuccessBackgroundColor: string;
  /**
   * @type {string}
   * @token Success-2
   * @description 提示框成功色边框颜色
   */
  alertSuccessBorderColor: string;
  /**
   * @type {string}
   * @token Success-6
   * @description 提示框成功色字体颜色
   */
  alertSuccessFontColor: string;
  /**
   * @type {string}
   * @token Warning-1
   * @description 提示框警告色背景颜色
   */
  alertWarningBackgroundColor: string;
  /**
   * @type {string}
   * @token Warning-2
   * @description 提示框警告色边框颜色
   */
  alertWarningBorderColor: string;
  /**
   * @type {string}
   * @token Warning-6
   * @description 提示框警告色字体颜色
   */
  alertWarningFontColor: string;
  /**
   * @type {string}
   * @token Danger-1
   * @description 提示框危险色背景颜色
   */
  alertDangerBackgroundColor: string;
  /**
   * @type {string}
   * @token Danger-2
   * @description 提示框危险色边框颜色
   */
  alertDangerBorderColor: string;
  /**
   * @type {string}
   * @token Danger-6
   * @description 提示框危险色字体颜色
   */
  alertDangerFontColor: string;
  /**
   * @type {string}
   * @token 14/regular
   * @description 提示框字体字号
   */
  alertFontSize: string;
  /**
   * @type {string}
   * @token Neutral-text-5
   * @description 提示框字体颜色
   */
  alertFontColor: string;
  /**
   * @type {string}
   * @token Padding-16
   * @description 提示框内边距（水平方向）
   */
  alertPaddingX: string;
  /**
   * @type {string}
   * @token Padding-7
   * @description 提示框内边距（垂直方向）
   */
  alertPaddingY: string;
  /**
   * @type {string}
   * @token Radius-4
   * @description 提示框边框圆角
   */
  alertBorderRadius: string;
  /**
   * @type {string}
   * @token Margin-8
   * @description 提示框undefined外边距
   */
  alertNearlyMargin: string;
  /**
   * @type {string}
   * @token 16/regular
   * @description 提示框标题模式字体字号
   */
  alertTitleFontSize: string;
  /**
   * @type {string}
   * @token Neutral-text-4
   * @description 提示框undefined字体颜色
   */
  alertCloseFontColor: string;
}
