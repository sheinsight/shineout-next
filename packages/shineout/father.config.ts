import { defineConfig } from 'father';
export default defineConfig({
  extends: '../../.fatherrc.ts',
});
