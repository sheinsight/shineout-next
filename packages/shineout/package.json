{
  "name": "shineout",
  "version": "3.0.0",
  "description": "A components library for React",
  "keywords": [],
  "license": "ISC",
  "author": "",
  "sideEffects": false,
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/index.js",
  "typings": "./dist/cjs/index.d.ts",
  "scripts": {
    "build": "father build",
    "build:watch": "father dev",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "dependencies": {
    "@sheinx/base": "workspace:*",
    "@sheinx/hooks": "workspace:*",
    "@sheinx/shineout-style": "workspace:*"
  },
  "peerDependencies": {
    "classnames": ">=2",
    "immer": ">=10",
    "react": ">=16.8"
  },
  "devDependencies": {
    "jszip": "^3.10.1"
  }
}
