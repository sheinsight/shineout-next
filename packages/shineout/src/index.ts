export { default as Alert } from './alert';
export { default as Button } from './button';
export { default as Checkbox } from './checkbox';
export { default as DatePicker } from './date-picker';
export { default as Dropdown } from './dropdown';
export { default as EditableArea } from './editable-area';
export { default as Form } from './form';
export { default as Icon } from './icon';
export { default as Image } from './image';
export { default as Input } from './input';
export { default as Popover } from './popover';
export { default as Radio } from './radio';
export { default as Rule } from './rule';
export { default as Spin } from './spin';
export { default as Switch } from './switch';
export { default as Tag } from './tag';
export { default as Tabs } from './tabs';
export { default as Textarea } from './textarea';
export { default as Tooltip } from './tooltip';
export { setConfig } from '@sheinx/base';

export * as TYPE from './type';
