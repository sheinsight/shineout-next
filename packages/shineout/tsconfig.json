{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,
    "baseUrl": "./",
    "paths": {
      "shineout": ["./src/index"],
      "@sheinx/base": ["../base/src/index"],
      "@sheinx/hooks": ["../hooks/src/index"]
    }
  },
  "include": ["src"]
}
