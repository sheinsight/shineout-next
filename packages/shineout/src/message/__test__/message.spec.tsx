import { render, cleanup, fireEvent, waitFor, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Message from '..';
import type { MessageOptions } from '../message.type';
import { Button } from 'shineout';
import {
  attributesTest,
  classContentTest,
  createClassName,
  delay,
  snapshotTest,
  styleTest,
  textContentTest,
} from '../../tests/utils';
import MessageBase from '../__example__/01-base';
import MessgaeDuration from '../__example__/02-duration';
import MessagePosition from '../__example__/03-position';
import MessageClose from '../__example__/04-close';
import MessageManualClose from '../__example__/05-manual-close';
import MessageContainer from '../__example__/06-container';

const SO_PREFIX = 'message';
const originClasses = ['wrapper', 'item', 'message'];
const originItemClasses = ['itemDismissed', 'itemShow'];
const { wrapper, item, itemShow, message } = createClassName(
  SO_PREFIX,
  originClasses,
  originItemClasses,
);
const SO_PREFIX_Alert = 'alert';
const originAlertClasses = ['content', 'close', 'title'];
const originAlertItemClasses = ['alert', 'info', 'success', 'warning', 'danger', 'pending'];
const {
  content: contentClassName,
  alert: alertClassName,
  close: closeClassName,
  title: titleClassName,
  ...rest
} = createClassName(SO_PREFIX_Alert, originAlertClasses, originAlertItemClasses);

const typeClassNameMap: { [key: string]: string } = {
  info: 'info',
  success: 'success',
  warn: 'warning',
  error: 'danger',
};

const MessageTest = ({ options }: { options?: MessageOptions }) => (
  <Button
    onClick={() => {
      Message.show('Test', 5, options);
    }}
  >
    Test
  </Button>
);

beforeAll(() => {
  jest.useFakeTimers();
});
afterAll(() => {
  jest.runAllTimers();
});

afterEach(cleanup);

describe('Message[Base]', () => {
  snapshotTest(<MessageBase />);
  snapshotTest(<MessgaeDuration />);
  snapshotTest(<MessagePosition />);
  test('should start with Shineout display', () => {
    expect(Message.displayName).toBe('ShineoutMessage');
  });
  test('should render default', async () => {
    const { container } = render(<MessageTest />);
    expect(document.querySelector(wrapper)).not.toBeInTheDocument();
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      const messageWrapper = document.querySelector(wrapper)!;
      attributesTest(messageWrapper, 'data-soui-position', 'top');
      const messageItem = messageWrapper.querySelector(item)!;
      classContentTest(messageItem, itemShow);
      const messageMain = messageItem.querySelector(message)!;
      classContentTest(messageMain, alertClassName);
      classContentTest(messageMain, rest.warning);
      textContentTest(messageMain.querySelector(contentClassName)!, 'Test');
      const messageClose = messageMain.querySelector(closeClassName)!;
      fireEvent.click(messageClose);
    });
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelector(wrapper)).not.toBeInTheDocument();
      fireEvent.click(container.querySelector('button')!);
    });
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelector(wrapper)).toBeInTheDocument();
    });
    await waitFor(
      async () => {
        await delay(3200);
        expect(document.querySelector(wrapper)).not.toBeInTheDocument();
      },
      { timeout: 10000 },
    );
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelectorAll(item).length).toBe(1);
    });
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelectorAll(item).length).toBe(2);
      expect(document.querySelector(wrapper)).toMatchSnapshot();
    });
    await waitFor(
      async () => {
        await delay(3200);
        expect(document.querySelectorAll(item).length).toBe(0);
      },
      { timeout: 10000 },
    );
  });
  test.each(['info', 'success', 'warn', 'error'])(
    'should render when set method is %s',
    async (type) => {
      const { container } = render(
        <Button
          onClick={() => {
            // @ts-ignore
            Message[type](`${type}`, 0);
          }}
        >
          Test
        </Button>,
      );
      fireEvent.click(container.querySelector('button')!);
      await waitFor(async () => {
        await delay(200);
        classContentTest(document.querySelector(message)!, rest[typeClassNameMap[type]]);
        fireEvent.click(document.querySelector(closeClassName)!);
      });
      await waitFor(async () => {
        await delay(200);
        expect(document.querySelector(wrapper)).not.toBeInTheDocument();
      });
    },
  );
  test('should render when set method is close', async () => {
    Message.success('success', 0);
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelectorAll(item).length).toBe(1);
    });
    Message.warn('warn', 0);
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelectorAll(item).length).toBe(2);
    });
    Message.close();
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelectorAll(item).length).toBe(0);
    });
  });
  test('should render when use setOptions', async () => {
    Message.setOptions({
      className: 'test',
    });
    const { container } = render(
      <Button
        onClick={() => {
          Message.show('Test', 0);
        }}
      >
        Test
      </Button>,
    );
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      classContentTest(document.querySelector(item)!, 'test');
      fireEvent.click(document.querySelector(closeClassName)!);
    });
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelector(wrapper)).not.toBeInTheDocument();
    });
  });
  test('should render when set duration', async () => {
    const { container, rerender } = render(
      <Button
        onClick={() => {
          Message.show('Test', 0);
        }}
      >
        Test
      </Button>,
    );
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelector(wrapper)).toBeInTheDocument();
    });
    await waitFor(
      async () => {
        await delay(3200);
        expect(document.querySelector(wrapper)).toBeInTheDocument();
      },
      { timeout: 10000 },
    );
    fireEvent.click(document.querySelector(closeClassName)!);
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelector(wrapper)).not.toBeInTheDocument();
    });
    rerender(
      <Button
        onClick={() => {
          Message.show('Test', 5);
        }}
      >
        Test
      </Button>,
    );
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelector(wrapper)).toBeInTheDocument();
    });
    await waitFor(
      async () => {
        await delay(5200);
        expect(document.querySelector(wrapper)).not.toBeInTheDocument();
      },
      { timeout: 10000 },
    );
  });
  test('should hover message when mouseenter', async () => {
    const { container } = render(<MessageTest />);
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelector(wrapper)).toBeInTheDocument();
      fireEvent.mouseEnter(document.querySelector(item)!);
    });
    await waitFor(
      async () => {
        await delay(3200);
        expect(document.querySelector(wrapper)).toBeInTheDocument();
        fireEvent.mouseLeave(document.querySelector(item)!);
      },
      { timeout: 10000 },
    );
    await waitFor(
      async () => {
        await delay(3200);
        expect(document.querySelector(wrapper)).not.toBeInTheDocument();
      },
      { timeout: 10000 },
    );
  });
});
describe('Message[Option]', () => {
  test('should render when set className', async () => {
    const { container } = render(
      <Button
        onClick={() => {
          Message.show('Test', 5, {
            className: 'demo',
          });
        }}
      >
        Test
      </Button>,
    );
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      classContentTest(document.querySelector(item)!, 'demo');
    });
    fireEvent.click(document.querySelector(closeClassName)!);
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelector(wrapper)).not.toBeInTheDocument();
    });
  });
  test('should render when set container', async () => {
    const { container } = render(<MessageContainer />);
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      expect(container.querySelector('#container')?.querySelector(wrapper)).toBeInTheDocument();
      fireEvent.click(container.querySelector(closeClassName)!);
    });
    await waitFor(async () => {
      await delay(200);
      expect(container.querySelector(wrapper)).not.toBeInTheDocument();
    });
  });
  test('should render when set hideClose is true', async () => {
    const { container } = render(<MessageTest options={{ hideClose: true }} />);
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelector(closeClassName)).not.toBeInTheDocument();
    });
  });
  test('should render when set onClose', async () => {
    const closeFn = jest.fn();
    const { container, rerender } = render(<MessageTest options={{ onClose: closeFn }} />);
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      fireEvent.click(document.querySelector(closeClassName)!);
    });
    await waitFor(async () => {
      await delay(200);
      expect(closeFn.mock.calls.length).toBe(1);
    });
    rerender(<MessageClose />);
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      fireEvent.click(document.querySelector(closeClassName)!);
    });
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelector(wrapper)).not.toBeInTheDocument();
    });
  });
  test('should render when set title', async () => {
    const { container } = render(<MessageTest options={{ title: 'hello' }} />);
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      textContentTest(document.querySelector(titleClassName)!, 'hello');
      fireEvent.click(document.querySelector(closeClassName)!);
    });
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelector(wrapper)).not.toBeInTheDocument();
    });
  });
  test('should render when set top', async () => {
    const { container } = render(<MessageTest options={{ top: '-20px' }} />);
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      styleTest(document.querySelector(message)!, 'top: -20px;');
      fireEvent.click(document.querySelector(closeClassName)!);
    });
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelector(wrapper)).not.toBeInTheDocument();
    });
  });
  test.each(['info', 'success', 'warn', 'error'])(
    'should render when set position is %s',
    async (type) => {
      const { container } = render(<MessageTest options={{ position: type }} />);
      fireEvent.click(container.querySelector('button')!);
      await waitFor(async () => {
        await delay(200);
        attributesTest(document.querySelector(wrapper)!, 'data-soui-position', type);
        fireEvent.click(document.querySelector(closeClassName)!);
      });
      await waitFor(async () => {
        await delay(200);
        expect(document.querySelector(wrapper)).not.toBeInTheDocument();
      });
    },
  );
  test('should render manual close', async () => {
    const { container } = render(<MessageManualClose />);
    fireEvent.click(container.querySelector('button')!);
    await waitFor(async () => {
      await delay(200);
      screen.debug();
      fireEvent.click(document.querySelectorAll('button')[1]);
    });
    await waitFor(async () => {
      await delay(200);
      expect(document.querySelector(wrapper)).not.toBeInTheDocument();
    });
  });
});
