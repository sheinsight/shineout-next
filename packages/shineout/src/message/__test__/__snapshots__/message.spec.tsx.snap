// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message[Base] should render correctly  1`] = `
<div>
  <button
    class="so-button-default so-button-button"
  >
    <span>
      Show
    </span>
  </button>
  <button
    class="so-button-default so-button-button"
  >
    <span>
      Info
    </span>
  </button>
  <button
    class="so-button-default so-button-button"
  >
    <span>
      Success
    </span>
  </button>
  <button
    class="so-button-default so-button-button"
  >
    <span>
      Warn
    </span>
  </button>
  <button
    class="so-button-default so-button-button"
  >
    <span>
      Error
    </span>
  </button>
  <button
    class="so-button-default so-button-button"
  >
    <span>
      Close All
    </span>
  </button>
</div>
`;

exports[`Message[Base] should render correctly  2`] = `
<div>
  <button
    class="so-button-default so-button-button"
  >
    <span>
      Duration 10 s.
    </span>
  </button>
  <button
    class="so-button-default so-button-button"
  >
    <span>
      Manually close
    </span>
  </button>
</div>
`;

exports[`Message[Base] should render correctly  3`] = `
<div>
  <div
    style="display: flex; align-items: center; margin-bottom: 24px;"
  >
    position:
    <div
      class="so-radio-group"
      style="margin: 0px 20px;"
    >
      <div
        class="so-radio-wrapper"
      >
        <input
          style="display: none;"
          tabindex="0"
          type="radio"
        />
        <span
          class="so-radio-indicator-wrapper"
        >
          <span
            class="so-radio-indicator"
            tabindex="1"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                clip-rule="evenodd"
                d="M12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3ZM12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1Z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </span>
        <span
          class="so-radio-desc"
        >
          top
        </span>
      </div>
      <div
        class="so-radio-wrapper"
      >
        <input
          style="display: none;"
          tabindex="0"
          type="radio"
        />
        <span
          class="so-radio-indicator-wrapper"
        >
          <span
            class="so-radio-indicator"
            tabindex="1"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                clip-rule="evenodd"
                d="M12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3ZM12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1Z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </span>
        <span
          class="so-radio-desc"
        >
          middle
        </span>
      </div>
      <div
        class="so-radio-wrapper"
      >
        <input
          style="display: none;"
          tabindex="0"
          type="radio"
        />
        <span
          class="so-radio-indicator-wrapper"
        >
          <span
            class="so-radio-indicator"
            tabindex="1"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                clip-rule="evenodd"
                d="M12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3ZM12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1Z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </span>
        <span
          class="so-radio-desc"
        >
          top-left
        </span>
      </div>
      <div
        class="so-radio-wrapper so-radio-wrapper-checked"
      >
        <input
          checked=""
          style="display: none;"
          tabindex="0"
          type="radio"
        />
        <span
          class="so-radio-indicator-wrapper"
        >
          <span
            class="so-radio-indicator"
            tabindex="1"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                clip-rule="evenodd"
                d="M3 12C3 7.02944 7.02944 3 12 3C16.9706 3 21 7.02944 21 12C21 16.9706 16.9706 21 12 21C7.02944 21 3 16.9706 3 12ZM1 12C1 5.92487 5.92487 1 12 1C18.0751 1 23 5.92487 23 12C23 18.0751 18.0751 23 12 23C5.92487 23 1 18.0751 1 12ZM12 7C9.23858 7 7 9.23858 7 12C7 14.7614 9.23858 17 12 17C14.7614 17 17 14.7614 17 12C17 9.23858 14.7614 7 12 7Z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </span>
        <span
          class="so-radio-desc"
        >
          top-right
        </span>
      </div>
      <div
        class="so-radio-wrapper"
      >
        <input
          style="display: none;"
          tabindex="0"
          type="radio"
        />
        <span
          class="so-radio-indicator-wrapper"
        >
          <span
            class="so-radio-indicator"
            tabindex="1"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                clip-rule="evenodd"
                d="M12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3ZM12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1Z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </span>
        <span
          class="so-radio-desc"
        >
          bottom-left
        </span>
      </div>
      <div
        class="so-radio-wrapper"
      >
        <input
          style="display: none;"
          tabindex="0"
          type="radio"
        />
        <span
          class="so-radio-indicator-wrapper"
        >
          <span
            class="so-radio-indicator"
            tabindex="1"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                clip-rule="evenodd"
                d="M12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3ZM12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1Z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </span>
        <span
          class="so-radio-desc"
        >
          bottom-right
        </span>
      </div>
    </div>
  </div>
  <button
    class="so-button-default so-button-button"
  >
    <span>
      Show message.
    </span>
  </button>
</div>
`;

exports[`Message[Base] should render default 1`] = `
<div
  class="so-message-wrapper"
  data-soui-position="top"
>
  <div
    class="so-message-item so-message-item-show "
  >
    <div
      class="so-message-message so-alert-alert so-alert-warning"
    >
      <div
        class="so-alert-content"
      >
        <div
          class="so-alert-text"
        >
          Test
        </div>
      </div>
      <div
        class="so-alert-close"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M4.6129 3.2097L4.70711 3.29289L12 10.585L19.2929 3.29289C19.6834 2.90237 20.3166 2.90237 20.7071 3.29289C21.0676 3.65338 21.0953 4.22061 20.7903 4.6129L20.7071 4.70711L13.415 12L20.7071 19.2929C21.0976 19.6834 21.0976 20.3166 20.7071 20.7071C20.3466 21.0676 19.7794 21.0953 19.3871 20.7903L19.2929 20.7071L12 13.415L4.70711 20.7071C4.31658 21.0976 3.68342 21.0976 3.29289 20.7071C2.93241 20.3466 2.90468 19.7794 3.2097 19.3871L3.29289 19.2929L10.585 12L3.29289 4.70711C2.90237 4.31658 2.90237 3.68342 3.29289 3.29289C3.65338 2.93241 4.22061 2.90468 4.6129 3.2097Z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="so-message-item so-message-item-show "
  >
    <div
      class="so-message-message so-alert-alert so-alert-warning"
    >
      <div
        class="so-alert-content"
      >
        <div
          class="so-alert-text"
        >
          Test
        </div>
      </div>
      <div
        class="so-alert-close"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M4.6129 3.2097L4.70711 3.29289L12 10.585L19.2929 3.29289C19.6834 2.90237 20.3166 2.90237 20.7071 3.29289C21.0676 3.65338 21.0953 4.22061 20.7903 4.6129L20.7071 4.70711L13.415 12L20.7071 19.2929C21.0976 19.6834 21.0976 20.3166 20.7071 20.7071C20.3466 21.0676 19.7794 21.0953 19.3871 20.7903L19.2929 20.7071L12 13.415L4.70711 20.7071C4.31658 21.0976 3.68342 21.0976 3.29289 20.7071C2.93241 20.3466 2.90468 19.7794 3.2097 19.3871L3.29289 19.2929L10.585 12L3.29289 4.70711C2.90237 4.31658 2.90237 3.68342 3.29289 3.29289C3.65338 2.93241 4.22061 2.90468 4.6129 3.2097Z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Message[Base] should render default 2`] = `
<div
  class="so-message-wrapper"
  data-soui-position="top"
>
  <div
    class="so-message-item so-message-item-show "
  >
    <div
      class="so-message-message so-alert-alert so-alert-warning"
    >
      <div
        class="so-alert-content"
      >
        <div
          class="so-alert-text"
        >
          Test
        </div>
      </div>
      <div
        class="so-alert-close"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M4.6129 3.2097L4.70711 3.29289L12 10.585L19.2929 3.29289C19.6834 2.90237 20.3166 2.90237 20.7071 3.29289C21.0676 3.65338 21.0953 4.22061 20.7903 4.6129L20.7071 4.70711L13.415 12L20.7071 19.2929C21.0976 19.6834 21.0976 20.3166 20.7071 20.7071C20.3466 21.0676 19.7794 21.0953 19.3871 20.7903L19.2929 20.7071L12 13.415L4.70711 20.7071C4.31658 21.0976 3.68342 21.0976 3.29289 20.7071C2.93241 20.3466 2.90468 19.7794 3.2097 19.3871L3.29289 19.2929L10.585 12L3.29289 4.70711C2.90237 4.31658 2.90237 3.68342 3.29289 3.29289C3.65338 2.93241 4.22061 2.90468 4.6129 3.2097Z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="so-message-item so-message-item-show "
  >
    <div
      class="so-message-message so-alert-alert so-alert-warning"
    >
      <div
        class="so-alert-content"
      >
        <div
          class="so-alert-text"
        >
          Test
        </div>
      </div>
      <div
        class="so-alert-close"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M4.6129 3.2097L4.70711 3.29289L12 10.585L19.2929 3.29289C19.6834 2.90237 20.3166 2.90237 20.7071 3.29289C21.0676 3.65338 21.0953 4.22061 20.7903 4.6129L20.7071 4.70711L13.415 12L20.7071 19.2929C21.0976 19.6834 21.0976 20.3166 20.7071 20.7071C20.3466 21.0676 19.7794 21.0953 19.3871 20.7903L19.2929 20.7071L12 13.415L4.70711 20.7071C4.31658 21.0976 3.68342 21.0976 3.29289 20.7071C2.93241 20.3466 2.90468 19.7794 3.2097 19.3871L3.29289 19.2929L10.585 12L3.29289 4.70711C2.90237 4.31658 2.90237 3.68342 3.29289 3.29289C3.65338 2.93241 4.22061 2.90468 4.6129 3.2097Z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
